src/cfg.rs

```diff
@@ -124,72 +123,88 @@ impl Default for Config {
     fn default() -> Self {
         Config {
             webserver: Some(WebServerConfig::default()),
-            log_level: "DEBUG".to_string(),
-            da_layer: DALayerOption::default(),
+            log_level: Some("DEBUG".to_string()),
+            da_layer: Some(DALayerOption::default()),
             celestia_config: Some(CelestiaConfig::default()),
             redis_config: Some(RedisConfig::default()),
-            epoch_time: 60,
+            epoch_time: Some(60),
             public_key: None,
         }
     }
 }
 
 pub fn load_config(args: CommandLineArgs) -> Result<Config, config::ConfigError> {
+    let config_path = args
+        .config_path
+        .unwrap_or_else(|| ".deimos/config.toml".to_string());
+
+    // if the config file doesn't exist, create it with the default values
+    if !Path::new(&config_path).exists() {
+        if let Some(parent) = Path::new(&config_path).parent() {
+            fs::create_dir_all(parent).unwrap();
+        }
+
+        let default_config = Config::default();
+        let config_toml = toml::to_string(&default_config).unwrap();
+        fs::write(&config_path, config_toml).unwrap();
+    }
+
     let settings = ConfigBuilder::<DefaultState>::default()
-        .add_source(File::from_str(
-            include_str!("config.toml"),
-            FileFormat::Toml,
-        ))
+        .add_source(File::with_name(&config_path))
         .build()?;
 
-    info!("{}", settings.get_string("log_level").unwrap_or_default());
-
     let default_config = Config::default();
+    let file_config: Config = settings.try_deserialize().unwrap_or_else(|e| {
+        debug!("Failed to deserialize config file: {}", e);
```

Comment:
```suggestion
        error!("deserializing config file: {}", e);
```

The rest of the logs everywhere are lowercase let's keep it consistent for aesthetic purposes
also, this is a valid error case, not a debug - and when we switch to error logging the "failed" part of the log is redundant